### ISSUE TITLE
New check: JavadocLinkWellKnownApi

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#19938]
# Description
After https://github.com/checkstyle/checkstyle/issues/19883

Javadoc comments should avoid unnecessary `{@link ...}` tags for APIs that are considered well-known. Linking well-known APIs can make comments harder to read without adding much value for the reader.

This check should report `{@link ...}` references to configured well-known APIs.

The check should support two properties:

- `wellKnownQualifiedPackages`: package names whose fully qualified API references should not be linked.
- `wellKnownSimpleNames`: simple API names that should not be linked.

Both properties are needed because Checkstyle does not resolve Javadoc link targets.
For example, `{@link java.lang.String}` contains the package name, so it can be matched through `wellKnownQualifiedPackages`.
However, `{@link String}` only contains the simple name `String`, so it needs to be matched through `wellKnownSimpleNames`. Resolution of imports is not a solution since `java.lang` is implicitly imported.

# Config

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocLinkWellKnownApi">
            <property name="wellKnownQualifiedPackages" value="java.lang"/>
            <property name="wellKnownSimpleNames" value="String"/>
        </module>
    </module>
</module>
```

### Default values:
- `wellKnownQualifiedPackages`: `java.lang`
- `wellKnownSimpleNames`: `String`

# Example
```java
public class Test {

    /**
     * Uses a String.
     */
    public String valid1() {
        return "";
    }

    /**
     * Uses a {@link String}. // violation
     */
    public String invalid1() {
        return "";
    }

    /**
     * Uses a {@link java.lang.String}. // violation
     */
    public String invalid2() {
        return "";
    }

    /**
     * Uses a {@link java.lang.Object}. // violation
     */
    public Object invalid3() {
        return new Object();
    }

    /**
     * Uses a {@link java.util.List}. // ok
     */
    public java.util.List<String> valid2() {
        return java.util.Collections.emptyList();
    }
}
```

# Output
```
[ERROR] Test.java:/:/: 'String' should not be linked because it is configured as a well-known API. [JavadocLinkWellKnownApi]
[ERROR] Test.java:/:/: 'java.lang.String' should not be linked because it belongs to a well-known package. [JavadocLinkWellKnownApi]
[ERROR] Test.java:/:/: 'java.lang.Object' should not be linked because it belongs to a well-known package. [JavadocLinkWellKnownApi]
Audit done.
Checkstyle ends with 3 errors.
```

**Comment by Carbon14-48** [comment]
hi @gianmarcoschifone  i want to work on this as well since the other one is waiting for review.  this check also should  flag
` {@linkplain ...}` tag right ?

**Comment by gianmarcoschifone** [comment]
> hi @gianmarcoschifone  i want to work on this as well since the other one is waiting for review.  this check also should  flag
> ` {@linkplain ...}` tag right ?

hi @Carbon14-48 , happy to see your interest. Yes also linkplain tag

**Comment by romani** [comment]
Checks is merged.

Time to activate in coverage report
